### ISSUE TITLE
iOS simulator 13.7 crashes with: dyld: lazy symbol binding failed: Symbol not found: _$s8StoreKit7ProductVMa

### ISSUE THREAD

**Issue by derRaab** [dooboolab/react-native-iap#1992]
**Description**

Using version `11.0.0-rc.5` in our app on an iOS simulator with version `13.7` does crash with this exception immediately after starting the app:

```
dyld: lazy symbol binding failed: Symbol not found: _$s8StoreKit7ProductVMa
  Referenced from: /Users/.../Library/Developer/CoreSimulator/Devices/8EB1F0D4-9E1E-407C-87BE-5B17DF1493CB/data/Containers/Bundle/Application/.../....app/...
  Expected in: /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 13.7.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/StoreKit.framework/StoreKit
```

Works fine with an iOS 16 simulator. :/

**Expected Behavior**

App doesn't crash.

**Environment:**

- react-native-iap: 11.0.0-rc.5
- react-native: 0.68.3
- Platforms iOS 13.7 simulator

**To Reproduce**

1. Install simulator 13.7
2. Run app on iOS 13.7

**Comment by ys-sherzad** [comment]
1+ 
Also happening in 14.+ versions

**Comment by andresesfm** [comment]
🤔 I think I might have to change the way Sk2 is being included: https://developer.apple.com/forums/thread/650886

**Comment by andresesfm** [comment]
Proposed fix is here: https://github.com/dooboolab/react-native-iap/pull/1994

**Comment by andresesfm** [comment]
Released as part of `11.0.0-rc.6`

**Comment by derRaab** [comment]
Hi @andresesfm - I tried this but now the iOS Simulator crashes hard around the time I call `getSubscriptions()` Since the app crashes I don't know exactly what happens because the Xcode console doesn't say much. BUT I get this sentry error in our logs:

```
EXC_BAD_INSTRUCTION: dealloc > priceLocale
  File "RNIapIos.swift", line 753, in RNIapIos.getDiscountData
  File "RNIapIos.swift", in RNIapIos.getProductObject
  File "<compiler-generated>", in RNIapIos.productsRequest
  File "<compiler-generated>", in thunk for closure
  File "<compiler-generated>", in thunk for closure
...
(13 additional frame(s) were not displayed)
```

<img width="476" alt="Bildschirmfoto 2022-09-24 um 18 17 09" src="https://user-images.githubusercontent.com/2079540/192108329-39c8a9e0-9cdb-4ff2-8ab7-8fa2272f9bed.png">

So I think this issue should not be closed. :)

**Comment by andresesfm** [comment]
We've battled with that Exception for a while but this is the first time I actually see someone posting the stacktrace. I haven't been able to reproduce locally. It might be related to this: https://stackoverflow.com/questions/53169358/pricelocale-throw-an-exc-breakpoint-when-accessed

**Comment by andresesfm** [comment]
Fixed as part of: `11.0.0-rc.7`
